service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/DiagnosticReportMapper.java

```diff
@@ -0,0 +1,184 @@
+
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+@Component
+@Slf4j
+public class DiagnosticReportMapper {
+
+    private static final Mustache DIAGNOSTIC_REPORT_COMPOUND_STATEMENT_TEMPLATE =
+        TemplateUtils.loadTemplate("diagnostic_report_compound_statement_template.mustache");
+
+    private final MessageContext messageContext;
+    private final SpecimenMapper specimenMapper;
+    private final ParticipantMapper participantMapper;
+    private final RandomIdGeneratorService randomIdGeneratorService;
+
+    public String mapDiagnosticReportToCompoundStatement(DiagnosticReport diagnosticReport) {
+        List<Specimen> specimens = fetchSpecimens(diagnosticReport);
+        List<Observation> observations = fetchObservations(diagnosticReport);
+
+        String diagnosticReportIssuedDate = DateFormatUtil.toHl7Format(diagnosticReport.getIssuedElement());
+
+        String mappedSpecimens = specimens.stream()
+            .map(specimen -> specimenMapper.mapSpecimenToCompoundStatement(specimen, observations, diagnosticReportIssuedDate))
+            .collect(Collectors.joining());
+
+        final IdMapper idMapper = messageContext.getIdMapper();
+
+        var diagnosticReportCompoundStatementTemplateParameters = DiagnosticReportCompoundStatementTemplateParameters.builder()
+            .compoundStatementId(idMapper.getOrNew(ResourceType.DiagnosticReport, diagnosticReport.getIdElement()))
+            .diagnosticReportIssuedDate(diagnosticReportIssuedDate)
+            .specimens(mappedSpecimens);
+
+        if (diagnosticReport.hasPerformer() && diagnosticReport.getPerformerFirstRep().hasActor()) {
+            final String participantReference = idMapper.get(diagnosticReport.getPerformerFirstRep().getActor());
+            final String participantBlock = participantMapper.mapToParticipant(participantReference, ParticipantType.AUTHOR);
+            diagnosticReportCompoundStatementTemplateParameters.participant(participantBlock);
+        }
+
+        StringBuilder reportLevelNarrativeStatements = new StringBuilder();
+
+        if (diagnosticReport.hasConclusion()) {
+            String narrativeStatementFromConclusion = buildNarrativeStatementForDiagnosticReport(
+                diagnosticReport, diagnosticReport.getConclusion()
+            );
+
+            reportLevelNarrativeStatements.append(narrativeStatementFromConclusion);
+        }
+
+        String codedDiagnosisText = diagnosticReport.getCodedDiagnosis().stream()
+            .map(this::extractTextFromCodedDiagnosisItems)
+            .flatMap(List::stream)
+            .collect(Collectors.joining(", "));
+
+        if (!codedDiagnosisText.isEmpty()) {
+            String narrativeStatementFromCodedDiagnosis = buildNarrativeStatementForDiagnosticReport(
+                diagnosticReport, codedDiagnosisText
+            );
+
+            reportLevelNarrativeStatements.append(narrativeStatementFromCodedDiagnosis);
+        }
```

Comment:
To codedDiagnosisText should be prepend with _Lab Diagnosis:_ 
and conclusion with _Interpretation:_ 

and I understand it that they both should be in one comment of type LABORATORY RESULT COMMENT(E141) , but it is something to be verified.

We need more tests to cover that cases.

Reply:
XML outputs of `conclusion` and `codedDiagnosis` will be in separate narrative statements. I'll update the branch to prepend those static texts